packages/types/src/span.ts

```diff
@@ -23,7 +23,13 @@ export type SpanAttributeValue =
   | Array<null | undefined | number>
   | Array<null | undefined | boolean>;
 
-export type SpanAttributes = Record<string, SpanAttributeValue | undefined>;
+export type SpanAttributes = Partial<{
+  'sentry.origin': string;
+  'sentry.op': string;
+  'sentry.source': string;
+  'sentry.sample_rate': number;
```

Comment:
I guess the downside is that we can't use the semantic attribute constants defined in `@sentry/core` (`as const` instead of the hard-coded names).

Reply:
I guess this is better than nothing, still! It's just some type sugar on top, so 🤷 still feels like an improvement!